lib/block_view_test.go

```diff
@@ -14690,3 +15303,1854 @@ func TestUpdateProfileChangeBack(t *testing.T) {
+			0,     /*MinBidAmountNanos*/
+		)
+		require.Error(err)
+		require.Contains(err.Error(), RuleErrorUpdateNFTByNonOwner)
+
+		// m1 trying to be sneaky by accepting their own bid.
+		_, _, _, err = _acceptNFTBid(
+			t, chain, db, params, 10,
+			m1Pub,
+			m1Priv,
+			post1Hash,
+			1, /*SerialNumber*/
+			m1Pub,
+			1,  /*BidAmountNanos*/
+			"", /*UnlockableText*/
+		)
+		require.Error(err)
+		require.Contains(err.Error(), RuleErrorAcceptNFTBidByNonOwner)
+	}
+
+	// Error case: accepting a bid that does not match the bid entry.
+	{
+		// m0 trying to be sneaky by setting m1's bid amount to 100x.
+		_, _, _, err = _acceptNFTBid(
+			t, chain, db, params, 10,
+			m0Pub,
+			m0Priv,
+			post1Hash,
+			1, /*SerialNumber*/
+			m1Pub,
+			100, /*BidAmountNanos*/
+			"",  /*UnlockableText*/
+		)
+		require.Error(err)
+		require.Contains(err.Error(), RuleErrorAcceptedNFTBidAmountDoesNotMatch)
+	}
+
+	// Error case: can't accept a non-existent bid.
+	{
+		// m3 has not bid on this NFT.
+		_, _, _, err = _acceptNFTBid(
+			t, chain, db, params, 10,
+			m0Pub,
+			m0Priv,
+			post1Hash,
+			1, /*SerialNumber*/
+			m3Pub,
+			200, /*BidAmountNanos*/
+			"",  /*UnlockableText*/
+		)
+		require.Error(err)
+		require.Contains(err.Error(), RuleErrorCantAcceptNonExistentBid)
+	}
+
+	// Error case: can't accept or update a non-existent NFT.
+	{
+		_, _, _, err := _updateNFT(
+			t, chain, db, params, 10,
+			m0Pub,
+			m0Priv,
+			post1Hash,
+			666,   /*SerialNumber*/
+			false, /*IsForSale*/
+			0,     /*MinBidAmountNanos*/
+		)
+		require.Error(err)
+		require.Contains(err.Error(), RuleErrorCannotUpdateNonExistentNFT)
+
+		_, _, _, err = _acceptNFTBid(
+			t, chain, db, params, 10,
+			m0Pub,
+			m0Priv,
+			post1Hash,
+			666, /*SerialNumber*/
+			m1Pub,
+			1,  /*BidAmountNanos*/
+			"", /*UnlockableText*/
+		)
+		require.Error(err)
+		require.Contains(err.Error(), RuleErrorNFTBidOnNonExistentNFTEntry)
+	}
+
+	// Error case: can't submit an update txn that doesn't actually update anything.
+	{
+		// <post1, #1> is already for sale.
+		_, _, _, err := _updateNFT(
+			t, chain, db, params, 10,
+			m0Pub,
+			m0Priv,
+			post1Hash,
+			1,    /*SerialNumber*/
+			true, /*IsForSale*/
+			0,    /*MinBidAmountNanos*/
+		)
+		require.Error(err)
+		require.Contains(err.Error(), RuleErrorNFTUpdateMustUpdateIsForSaleStatus)
+	}
+
+	// Finally, accept m2's bid on <post1, #1>.
+	{
+		_acceptNFTBidWithTestMeta(
+			testMeta,
+			10, /*FeeRateNanosPerKB*/
+			m0Pub,
+			m0Priv,
+			post1Hash,
+			1, /*SerialNumber*/
+			m2Pub,
+			2,  /*BidAmountNanos*/
+			"", /*UnlockableText*/
+		)
+	}
+
+	// Update <post1, #2>, so that it is no longer for sale.
+	{
+		_updateNFTWithTestMeta(
+			testMeta,
+			10, /*FeeRateNanosPerKB*/
+			m0Pub,
+			m0Priv,
+			post1Hash,
+			2,     /*SerialNumber*/
+			false, /*IsForSale*/
+			0,     /*MinBidAmountNanos*/
+		)
+	}
+
+	// Error case: <post1, #1> and <post1, #2> are no longer for sale and should not accept bids.
```

Comment:
should add a test for should not be able to AcceptNFTBid if nft is not for sale.

Reply:
and maybe we redo this comment to be `should not allow new bids` since we use the term "accept" to mean users accepts a bid for an NFT thus transferring to it the bidder.

Reply:
See "TestNFTMoreErrorCases"